test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -18,39 +18,55 @@
 
 #include <array>
 #include <map>
+#include <memory>
 #include <string>
+#include <utility>
+#include <vector>
 
 #include "absl/status/status.h"
+#include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "absl/types/span.h"
 #include "gtest/gtest.h"
 
 #include <grpc/grpc.h>
 
 #include "src/core/ext/filters/stateful_session/stateful_session_filter.h"
+#include "src/core/ext/xds/xds_health_status.h"
 #include "src/core/lib/gprpp/orphanable.h"
 #include "src/core/lib/gprpp/ref_counted_ptr.h"
 #include "src/core/lib/gprpp/unique_type_name.h"
 #include "src/core/lib/json/json.h"
 #include "src/core/lib/load_balancing/lb_policy.h"
+#include "src/core/lib/resolver/server_address.h"
 #include "test/core/client_channel/lb_policy/lb_policy_test_lib.h"
 #include "test/core/util/test_config.h"
 
 namespace grpc_core {
 namespace testing {
 namespace {
+using AddressAttributes =
+    std::map<const char*, std::unique_ptr<ServerAddress::AttributeInterface>>;
 
 class XdsOverrideHostTest : public LoadBalancingPolicyTest {
  protected:
   XdsOverrideHostTest()
       : policy_(MakeLbPolicy("xds_override_host_experimental")) {}
 
   RefCountedPtr<LoadBalancingPolicy::Config> MakeXdsOverrideHostConfig(
+      bool exclude_draining_status = false,
```

Comment:
Instead of using a bool here, I suggest passing in the set of override host statuses, since we will need tests to verify that we honor the override host status set even for non-draining statuses -- e.g., if the override host status set includes only HEALTHY, then the picker needs to ignore hosts with status UNKNOWN.

Please make sure that for each of the three health statuses, we have two tests: one that shows that an override host with the status will be used if it is present in the override host status set, and another that shows that an override host with the status will *not* be used if it is *not* present in the override host status set.